libs/core/tests/unit/string_tests.cpp

```diff
@@ -51,4 +62,85 @@ TEST(StringTests, check_StartsWith)
   EXPECT_FALSE(StartsWith("Hello World", "...Hello World"));
 }
 
+using namespace fetch::string;
+
+TEST(StringTests, check_Replace)
+{
+  EXPECT_EQUAL(Replace("Space shuttle ready to start", 's', 'z'), "Space zhuttle ready to ztart");
+  EXPECT_EQUAL(Replace("Space shuttle ready to start", 'z', 'm'), "Space shuttle ready to start");
+}
+
+TEST(StringTests, check_Trim)
+{
+  std::string s;
+
+  // TrimFromRight
+  s = "    1234";
+  TrimFromRight(s);
+  EXPECT_EQ(s, "1234");
+
+  s = "1234";
+  TrimFromRight(s);
+  EXPECT_EQ(s, "1234");
+
+  s = "    ";
+  TrimFromRight(s);
+  EXPECT_EQ(s, "");
+
+  // TrimFromLeft
```

Comment:
Rather than add these comments, I think it would be neater to split this test case into three, one for left, right and two-sided trim each.